liquidity/lib/useUndelegateBaseAndromeda/useUndelegateBaseAndromeda.tsx

```diff
@@ -45,22 +45,29 @@ export function useUndelegateBaseAndromeda({ collateralChange }: { collateralCha
   const { data: DebtRepayer } = useDebtRepayer();
 
   const queryClient = useQueryClient();
+
+  const canUndelegate =
+    liquidityPosition &&
+    liquidityPosition.collateralAmount.gt(0) &&
+    undelegateAmount &&
+    liquidityPosition.collateralAmount.gte(undelegateAmount);
+
+  const isReady =
+    canUndelegate &&
+    undelegateAmount.gt(0) &&
+    network &&
+    provider &&
+    signer &&
+    CoreProxy &&
+    AccountProxy &&
+    DebtRepayer &&
+    SpotMarketProxy &&
+    params.accountId &&
+    !!collateralType?.tokenAddress;
```

Comment:
```suggestion
    collateralType;
```
And wrap whole thing in Boolean() please